pallets/parachain-staking/src/lib.rs

```diff
@@ -502,7 +550,11 @@ pub mod pallet {
 					collator_count,
 					total_staked,
 				));
-			}
+				(collator_count, nomination_count)
+			} else {
+				(0u32, 0u32)
+			};
+			T::WeightInfo::on_initialize(collator_count, nomination_count)
```

Comment:
The `on_initialize` weight depends on the number of collators, number of total nominations.

A good question would be, why isn't `execute_delayed_collator_exits()` included in this weight calculation. The weight is charged to the caller of `leave_candidates` when they make the call to schedule their collator's exit.

Reply:
Needs to also include collator exits in this computation. See comment further below, more TODO for me!